viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -111,52 +111,65 @@ export class TraceViewerWidget extends ReactWidget {
 
                     const traces = new Array<Trace>();
                     if (isCancelled.value) {
-                        progress.report({message: 'Complete', work: {done: 100, total: 100}});
+                        progress.report({ message: 'Complete', work: { done: 100, total: 100 } });
                         this.dispose();
                         return;
                     }
 
-                    progress.report({message: 'Opening traces', work: {done: 30, total: 100}});
+                    progress.report({ message: 'Opening traces', work: { done: 30, total: 100 } });
+                    const invalidTraces = new Array<string>();
                     for (let i = 0; i < tracesArray.length; i++) {
                         if (isCancelled.value) {
                             break;
                         }
                         const trace = await this.traceManager.openTrace(tracesArray[i].toString(), tracesArray[i].name + tracesArray[i].ext);
                         if (trace) {
                             traces.push(trace);
+                        } else {
+                            invalidTraces.push(tracesArray[i].name.concat(tracesArray[i].ext));
                         }
                     }
 
                     if (isCancelled.value) {
                         // Rollback traces
-                        progress.report({message: 'Rolling back traces', work: {done: 50, total: 100}});
+                        progress.report({ message: 'Rolling back traces', work: { done: 50, total: 100 } });
                         for (let i = 0; i < traces.length; i++) {
                             await this.traceManager.closeTrace(traces[i].UUID);
                         }
-                        progress.report({message: 'Complete', work: {done: 100, total: 100}});
+                        progress.report({ message: 'Complete', work: { done: 100, total: 100 } });
                         this.dispose();
                         return;
                     }
-                    progress.report({message: 'Merging traces', work: {done: 70, total: 100}});
-                    const experiment = await this.experimentManager.openExperiment(this.uri.name + this.uri.ext, traces);
-                    if (experiment) {
-                        this.openedExperiment = experiment;
-                        this.title.label = 'Trace: ' + experiment.name;
-                        this.id = experiment.UUID;
+                    progress.report({ message: 'Merging traces', work: { done: 70, total: 100 } });
+
+                    if (traces === undefined || traces.length === 0) {
+                        // All the traces are invalid. Display the error message and exit.
+                        this.messageService.error('Invalid trace(s): ' + invalidTraces.toString());
```

Comment:
When trying to import a folder that contains no metadata files, the message will say "Invalid trace(s): ", which is weird. In case of `invalidTraces.length == 0`  can we display something along the lines of "No valid traces found in the selected directory"?

Reply:
> When trying to import a folder that contains no metadata files, the message will say "Invalid trace(s): ", which is weird. In case of `invalidTraces.length == 0` can we display something along the lines of "No valid traces found in the selected directory"?


Good catch, I have added a check to see if the folder is empty (doesn't contain metadata files). 

Note: There are other corner cases that we need to address, post MVP, for example (what happens if you upload a JSON file, txt file or other non-ctf traces.